### PR TITLE
Format short blocks and loops on a single line.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libyul/AsmPrinter.cpp [hunk 1]
```diff
@@ -212,15 +216,19 @@ string AsmPrinter::operator()(Switch const& _switch) const
...
+	string post = (*this)(_forLoop.post);
+	char delim = '\n';
+	if (
+		pre.size() + condition.size() + post.size() < 60 &&
+		pre.find('\n') == string::npos &&
+		post.find('\n') == string::npos
+	)
+		delim = ' ';
+	return
+		("for " + move(pre) + delim + move(condition) + delim + move(post) + "\n") +
```
Comment (ekpyron) [inline libyul/AsmPrinter.cpp:230]: Moving doesn't help for concatenation at all - it needs to be copied anyways, right?
One could also think about benchmarking string concatenation - I reckon that a custom concatenation function that creates an empty string, ``reserve()``s the required space and then adds the elements with ``append`` will be fastest, but that's independent of this PR.

**Review by ekpyron — DISMISSED** [review]
Looks good despite the ``move``s probably being in vain.

**Comment by ekpyron** [comment]
Yeah, ok, I was actually wondering why it's not more tests and there are indeed failing ones.

File: libyul/AsmPrinter.cpp (hunk 1 above)
Comment (chriseth) [reply libyul/AsmPrinter.cpp:230]: There are rvalue overloads for `operator+` for strings. It migh not make too much sense in this case, but `move(x) + move(y)` is certainly faster than `x+y` in the case that `x` does not have to be resized.

I benchmarked the use of `reserve` some years back and could not find any speed benefit.

**Comment by chriseth** [comment]
Oh right! All the optimizer tests...

**Comment by chriseth** [comment]
600 deleted lines! Not bad ;)

**Review by ekpyron — APPROVED** [review]
I just quickly scrolled through the updated optimizer tests and it looks reasonable, so I'd just trust isoltest here :-).

File: libyul/AsmPrinter.cpp (hunk 1 above)
Comment (ekpyron) [reply libyul/AsmPrinter.cpp]: Ok - well the move doesn't hurt in any case and maybe ``"for " + move(pre)`` will even use ``pre``s space if that suffices.
I just read some  benchmarks suggesting that ``reserve`` is fastest, though - maybe one should check that again - but in any case nothing for this PR.

**Pull request merged**